core/src/main/java/com/coderanch/blackjack/MiniGame.java

```diff
@@ -0,0 +1,186 @@
+
+    /**
+     * The deck of cards used in the game.
+     */
+    private final Deque<Card> deck = new ArrayDeque<>(Cards.getShuffledStandardDeck(new SecureRandom()));
+
+    /**
+     * The input utility used for getting answers.
+     */
+    private final InputUtility inputUtility;
+
+    /**
+     * The print stream used for sending text to the user.
+     */
+    private final PrintStream printStream;
+
+    /**
+     * Represents whether the user has passed or not.
+     */
+    private boolean isPassed;
+
+    /**
+     * Constructs new mini game of Blackjack.
+     *
+     * @param inputUtility used for user input.
+     * @param printStream used for printing text output.
+     */
+    public MiniGame(InputUtility inputUtility, PrintStream printStream) {
+        requireThat("inputUtility", inputUtility, is(notNullValue()));
+        requireThat("printStream", printStream, is(notNullValue()));
+        this.inputUtility = inputUtility;
+        this.printStream = printStream;
+    }
+
+    /**
+     * Run the mini game.
+     * Will start a game of Blackjack.
+     * Will ask the user to hit or pass until he wins, loses, or passes.
+     *
+     * @throws IOException if an I/O exception occurred while prompting the user for an action.
+     */
+    public void run() throws IOException {
+        hand = new Hand(deck.removeLast(), deck.removeLast());
+        displayStatus();
+
+        while (!isGameOver()) {
+            var choice = getUserChoice();
+            switch (choice) {
+                case HIT:
+                    dealCard();
+                    displayStatus();
+                    break;
+
+                case PASS:
+                default:
+                    pass();
+                    break;
+            }
+        }
+    }
+
+    /**
+     * Get the user's next choice.
+     *
+     * @return the user's choice.
+     * @throws IOException if there's a problem with the underlying stream.
+     */
+    private CHOICE getUserChoice() throws IOException {
+        var result = inputUtility.nextString("hit or pass?", PLAY_PREDICATE);
+        return Enum.valueOf(CHOICE.class, result.trim().toUpperCase(Locale.ENGLISH));
+    }
+
+    /**
+     * Remember that the user has passed.
+     */
+    private void pass() {
+        isPassed = true;
+    }
+
+    /**
+     * Display the status of the game.
+     */
+    private void displayStatus() {
+        printStream.println("Your cards are: ");
+        hand.cards().forEach(printStream::println);
+        printStream.println(String.format("Your score is: %d", hand.bestScore()));
+    }
+
+    /**
+     * Checks to see if the game is over.
+     *
+     * @return if the game is over or not.
+     */
+    private boolean isGameOver() {
+        if (hand.bestScore() == MAX_LEGAL_SCORE) {
+            printStream.println("You win. Game over.");
+            return true;
+        }
+        if (hand.isBust()) {
+            printStream.println("You lose. Game over.");
```

Comment:
I don't really expect printing statements in a method whose main purpose is to return a simple property of the object.

Move them to the `run()` method.